packages/notifications/src/elements/toaster/ToastSlot.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { ThemeContext } from 'styled-components';
+import { CSSTransition } from 'react-transition-group';
+import { useDocument } from '@zendeskgarden/react-theming';
+import { IToast, ToastPlacement } from './reducer';
+import { Toast } from './Toast';
+import { StyledFadeInTransition, StyledTransitionGroup, TRANSITION_CLASS } from './styled';
+
+interface IToastSlotProps {
+  toasts: IToast[];
+  placement: ToastPlacement;
+  limit: number;
+  zIndex?: number;
+}
+
+export const ToastSlot: React.FC<IToastSlotProps> = ({ toasts, placement, zIndex, limit }) => {
+  const [pauseTimers, setPauseTimers] = useState(false);
+  const theme = useContext(ThemeContext);
+  const environment = useDocument(theme);
+
+  const handleVisibilityChange = useCallback((e: Event) => {
+    if ((e.target as Document).visibilityState === 'visible') {
+      setPauseTimers(false);
+    } else {
+      setPauseTimers(true);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (environment) {
+      environment.addEventListener('visibilitychange', handleVisibilityChange);
+    }
+
+    return () => {
+      if (environment) {
+        environment.removeEventListener('visibilitychange', handleVisibilityChange);
+      }
+    };
+  }, [environment, handleVisibilityChange]);
+
+  const handleMouseEnter = useCallback(() => {
+    setPauseTimers(true);
+  }, []);
+
+  const handleMouseLeave = useCallback(() => {
+    setPauseTimers(false);
+  }, []);
+
+  const isHidden = useCallback(
+    (index: number) => {
+      if (placement === 'bottom' || placement === 'bottom-start' || placement === 'bottom-end') {
+        return index < toasts.length - limit;
+      }
+
+      return index >= limit;
+    },
+    [limit, placement, toasts.length]
+  );
+
+  return (
+    <StyledTransitionGroup
+      key={placement}
+      placement={placement}
+      zIndex={zIndex}
```

Comment:
I noticed a warning about passing `zIndex` to the native dom element in the console while running the branch locally:

<img width="1811" alt="Screen Shot 2021-02-26 at 11 31 55 AM" src="https://user-images.githubusercontent.com/1811365/109347289-c7701080-7827-11eb-9bc3-af57181509dd.png">

Reply:
These components might be a good first candidate for [transient style props](https://styled-components.com/docs/api#transient-props)

Reply:
I've updated these props to use the `$*` transient naming. We will still see this issues until we upgrade to v5, but it will be resolved at that time.